src/Data/Eq.purs

```diff
@@ -18,6 +18,9 @@ import Data.Void (Void)
 -- | class due to the presence of `NaN`, since `NaN /= NaN`. Additionally,
 -- | computing with `Number` can result in a loss of precision, so sometimes
 -- | values that should be equivalent are not.
+-- |
+-- | It is generally recommended to declare `Eq` instances where
+-- | `x == y && f x == f y` but not a strict requirement.
```

Comment:
Should this `&&` be `implies`?

Also perhaps this should be more explicit that the two `==` are potentially using different instances.

Is it worth giving a short example where this isn't true? Like:
```
newtype Identical a = Identical a
unidentical (Identical a) = a
instance eqIdentical :: Eq (Identical a) where eq _ _ = true
```
So this `Eq` instance isn't isomorphic to the `Eq` of the underlying type.  Therefore, `unidentical` is a function that lets you recover the distinction between values.  If `f ~ unidentical` then ```x == y `implies` f x == f y``` is false.

Reply:
And then we'd possibly need to say what kind of injectivity we're talking about, because if it's with respect to `Eq`, then `Identical`/`unidentical` aren't valid `f`s -  because the `Eq` isn't iso to the inner type's `Eq`.  That is, the equality semantics need to be talking "in the same language" to make sense.
If it's injective with respect to definitional equality, then those functions are injective. But that messes things up 😅.

Perhaps a less wordy way to say all this is:
```
An `Eq t` instance should ideally be consistent with the `Eq` instances of everything that can be injected into and projected out of `t`, should those instances exist.
```

Reply:
Kmett said it better: "structural and reflexive" 😸 
https://www.reddit.com/r/haskell/comments/87mf4v/is_it_considered_acceptable_for_eq_instances_to/